src/Chapter3.hs

```diff
@@ -971,6 +1110,22 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday
+    deriving (Show, Enum, Eq, Bounded)
+
+isWeekend :: Weekday -> Bool
+isWeekend Sunday   = True
+isWeekend Saturday = True
+isWeekend _        = False
+
+nextDay :: Weekday -> Weekday
+nextDay day
+    | day == maxBound = minBound
+    | otherwise       = succ day
```

Comment:
Your implementation is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂